Python/_warnings.c

```diff
@@ -254,6 +254,87 @@ warnings_lock_held(WarningsState *st)
     return PyMutex_IsLocked(&st->lock.mutex);
 }
 
+static PyObject *
+get_warnings_context(PyInterpreterState *interp)
+{
+    PyObject *ctx_var = GET_WARNINGS_ATTR(interp, _warnings_context, 0);
```

Comment:
How about creating `_warnings_context` contextvar as a state variable in C implementation then override the pure python implementation with it? That would make working in C implementation much easier.

Reply:
That's a good idea.  So much so that I already tried to do that.  :-)   I ran into difficulties making the tests work.  They import the modules two times in order to test both the Python and C versions of the module.  I'm not a fan of how those tests are organized so maybe I should just re-write a bunch of things there.  I was trying to avoid making too many changes.

Making `_warnings_context` a variable inside the C extension would be nicer.  We can't do that with `warnings.filters` because user code assigns it.  But we could with `_warnings_context`.  It can be done and it's cleaner (probably better performing too) and I'll try again.

Reply:
It took some extra modifications to `warnings.py` but I got this to work.  `_warnings` now has a `_warnings_context` attribute and that's used by the C extension, rather than having to look into `sys.modules['warnings]` to get it.